code/game/machinery/cryopod.dm

```diff
@@ -77,310 +123,268 @@ var/global/list/frozen_items = list("Alpha"=list(),"Bravo"=list(),"Charlie"=list
 
 /obj/machinery/cryopod/right
-	orient_right = 1
+	orient_right = TRUE
 	icon_state = "body_scanner_0-r"
 
-/obj/machinery/cryopod/New()
+/obj/machinery/cryopod/Initialize()
+	. = ..()
+	announce = new(src)
+	update_icon()
 
-	announce = new /obj/item/device/radio/intercom(src)
+/obj/machinery/cryopod/update_icon()
+	var/occupied = occupant ? TRUE : FALSE
+	var/mirror = orient_right ? "-r" : ""
+	icon_state = "body_scanner_[occupied][mirror]"
 
-	if(orient_right)
-		icon_state = "body_scanner_0-r"
+/obj/machinery/cryopod/process()
+	if(!occupant)
+		stop_processing()
+		update_icon()
+		return
+
+	if(occupant.stat == DEAD) //Occupant is dead, abort.
+		go_out()
+		return
+
+	//Allow a ten minute gap between entering the pod and actually despawning.
+	if(world.time - time_entered < time_till_despawn)
+		return
+
+	occupant.despawn(src)
+	stop_processing()
+	update_icon()
+
+/mob/proc/despawn(obj/machinery/cryopod/pod, dept_console = CRYO_REQ)
+
+	//Handle job slot/tater cleanup.
+	if(job)
+		var/datum/job/J = SSjob.name_occupations[job]
+		J.current_positions--
+		if((J.title in JOBS_REGULAR_ALL) && isdistress(SSticker?.mode))
+			var/datum/game_mode/distress/D = SSticker.mode
+			D.latejoin_tally-- //Cryoing someone removes a player from the round, blocking further larva spawns until accounted for
+		if(J.title in JOBS_POLICE)
+			dept_console = CRYO_SEC
+		else if(J.title in JOBS_MEDICAL)
+			dept_console = CRYO_MED
+		else if(J.title in JOBS_ENGINEERING)
+			dept_console = CRYO_ENGI
+
+	var/list/stored_items = list()
+	for(var/obj/item/W in src)
+		stored_items.Add(W.store_in_cryo())
+	GLOB.cryoed_item_list[dept_console].Add(stored_items)
+
+
+	for(var/datum/data/record/R in GLOB.datacore.medical)
+		if((R.fields["name"] == real_name))
+			GLOB.datacore.medical -= R
+			qdel(R)
+	for(var/datum/data/record/T in GLOB.datacore.security)
+		if((T.fields["name"] == real_name))
+			GLOB.datacore.security -= T
+			qdel(T)
+	for(var/datum/data/record/G in GLOB.datacore.general)
+		if((G.fields["name"] == real_name))
+			GLOB.datacore.general -= G
+			qdel(G)
+
+	ghostize(FALSE) //We want to make sure they are not kicked to lobby.
+
+	//Make an announcement and log the person entering storage.
+	var/data = num2text(length(GLOB.cryoed_mob_list))
+	GLOB.cryoed_mob_list += data
+	GLOB.cryoed_mob_list[data] = list(real_name, job ? job : "Unassigned", gameTimestamp())
+
+	var/obj/item/device/radio/intercom/radio
+	if(pod)
+		radio = pod.announce
+		pod.visible_message("<span class='notice'>[pod] hums and hisses as it moves [real_name] into hypersleep storage.</span>")
+		pod.occupant = null
 	else
-		icon_state = "body_scanner_0"
-	..()
+		radio = new(src)
+	radio.autosay("[real_name] has entered long-term hypersleep storage. Belongings moved to hypersleep inventory.", "Hypersleep Storage System")
+
+	qdel(src)
+
+/mob/living/carbon/human/despawn(obj/machinery/cryopod/pod, dept_console = CRYO_REQ)
+	if(assigned_squad)
+		var/datum/squad/S = assigned_squad
```

Comment:
The `assigned_squad` mob var is already type cast, unlike the mind one. No need to do it again.